t/apicast-policy-headers.t

```diff
@@ -0,0 +1,510 @@
+      end
+    }
+  }
+--- upstream
+  location / {
+     content_by_lua_block {
+       if ngx.req.get_headers()['New-Header'] ~= 'config_value_nh' then
+         ngx.log(ngx.ERR, "The header 'New-Header' does not have the correct value")
+       elseif ngx.req.get_headers()['Existing-Header'] ~= 'config_value_eh' then
+         ngx.log(ngx.ERR, "The header 'Existing-Header' does not have the correct value")
+       elseif ngx.req.get_headers()['Header-To-Delete'] then
+         ngx.log(ngx.ERR, "Received unexpected 'Header-To-Delete' header")
+       else
+         ngx.say('yay, api backend');
+       end
+     }
+  }
+--- request
+GET /?user_key=value
+--- more_headers
+Existing-Header: request_value_eh1
+Existing-Header: request_value_eh2
+Header-To-Delete: request_value_htd
+--- response_body
+yay, api backend
+--- response_headers
+New-Header:
+Existing-Header:
+Header-To-Delete:
+--- error_code: 200
+--- no_error_log
+[error]
+
+=== TEST 2: 'add' operation in request headers
+We test 3 things:
+1) Add op with a header that does not exist creates it with the given value.
+2) Add op with a header that exists, creates a new header with the same name
+   and the given value.
+3) Check that the headers are received in the upstream but not in the response
+   of the original request.
+--- configuration
+{
+  "services": [
+    {
+      "id": 42,
+      "backend_version":  1,
+      "backend_authentication_type": "service_token",
+      "backend_authentication_value": "token-value",
+      "proxy": {
+        "policy_chain": [
+          { "name": "apicast.policy.apicast" },
+          {
+            "name": "apicast.policy.headers",
+            "configuration":
+              {
+                "request":
+                  [
+                    { "op": "add", "header": "New-Header", "value": "config_value_nh" },
+                    { "op": "add", "header": "Existing-Header", "value": "config_value_eh" }
+                  ]
+              }
+          }
+        ],
+        "api_backend": "http://test:$TEST_NGINX_SERVER_PORT/",
+        "proxy_rules": [
+          { "pattern": "/", "http_method": "GET", "metric_system_name": "hits", "delta": 2 }
+        ]
+      }
+    }
+  ]
+}
+--- backend
+  location /transactions/authrep.xml {
+    content_by_lua_block {
+      local expected = "service_token=token-value&service_id=42&usage%5Bhits%5D=2&user_key=value"
+      local args = ngx.var.args
+      if args == expected then
+        ngx.exit(200)
+      else
+        ngx.log(ngx.ERR, expected, ' did not match: ', args)
+        ngx.exit(403)
+      end
+    }
+  }
+--- upstream
+  location / {
+     content_by_lua_block {
+       local existing_header_values = ngx.req.get_headers()['Existing-Header']
+
+       if ngx.req.get_headers()['New-Header'] ~= 'config_value_nh' then
+         ngx.log(ngx.ERR, "The header 'New-Header' does not have the correct value")
+       elseif existing_header_values[1] ~= 'request_value_eh1' or
```

Comment:
Should use use `assert.same` here ? 

It should be easy as `assert = require("luassert")`.

Reply:
Yes. Good idea :+1: